lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -2706,6 +2722,9 @@ constructTransaction api argGenChange knownPools poolStatus apiWalletId body = d
                 } -> liftHandler $ throwE ErrConstructTxWrongPayload
             _ -> pure ()
 
+    when (isJust (body ^. #vote)) $
+        liftHandler $ W.votingEraValidation nl
```

Comment:
Please use the era from the existing `W.readNodeTipStateForTxWrite` call to perform the validation instead of adding `era <- currentNodeEra nw` — to avoid having two sources of truth with potential race conditions